src/datasets/iterable_dataset.py

```diff
@@ -1076,6 +1114,12 @@ def _iter(self):
                 if self.remove_columns:
                     for c in self.remove_columns:
                         del transformed_example[c]
+                if self._return_features:
+                    for c in self._return_features.keys():
+                        if c not in transformed_example:
+                            transformed_example[c] = None
+                    # TODO: check types
+                    transformed_example = self._return_features.decode_example(transformed_example)
```

Comment:
is it expected that it is only applied to the unbatched case ?

Reply:
nope! fixed thanks for spotting